sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+                            String authStringEnc = Base64.getEncoder().encodeToString(authString.getBytes());
+                            connection.setRequestProperty("Proxy-Authorization", "Basic " + authStringEnc);
+                        }
+                    }
+                } else {
+                    connection = (HttpURLConnection) url.openConnection();
+                }
+
+                if (connectionTimeout != null) {
+                    connection.setConnectTimeout((int) connectionTimeout.toMillis());
+                }
+
+                // SetConnectionRequest
+                try {
+                    connection.setRequestMethod(httpRequest.getHttpMethod().toString());
+                } catch (ProtocolException e) {
+                    return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                }
+                for (HttpHeader header : httpRequest.getHeaders()) {
+                    for (String value : header.getValues()) {
+                        connection.addRequestProperty(header.getName(), value);
+                    }
+                }
+                if (httpRequest.getHttpMethod() != HttpMethod.GET) {
+                    connection.setDoOutput(true);
+                }
+
+                // Write body
+                switch (httpRequest.getHttpMethod()) {
+                    case POST:
+                    case PUT:
+                    case DELETE:
+                        if (httpRequest.getBody() != null) {
+                            try (DataOutputStream os = new DataOutputStream(new BufferedOutputStream(connection.getOutputStream()))) {
+
+                                Flux<ByteBuffer> requestBody = httpRequest.getBody();
+                                if (progressReporter != null) {
+                                    requestBody = requestBody.map(buffer -> {
+                                        progressReporter.reportProgress(buffer.remaining());
+                                        return buffer;
+                                    });
+                                }
+
+                                requestBody
+                                    .flatMap(buffer -> {
+                                        try {
+                                            byte[] bytes = new byte[buffer.remaining()];
+                                            buffer.get(bytes);
+                                            os.write(bytes);
+                                            return Mono.just(buffer); // Emit the buffer for downstream processing if needed
+                                        } catch (IOException e) {
+                                            return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                                        }
+                                    })
+                                    .then()
+                                    .block(); // Wait for completion of the write operations
```

Comment:
This shouldn't `block` here, instead this can be chained with receive response code below.